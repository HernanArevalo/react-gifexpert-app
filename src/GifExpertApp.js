import React, { useState } from 'react'
import { AddCattegory } from './components/AddCattegory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = ['One Punch']}) => {

    // const [categories, setCategories] = useState(['One Punch'])
    const [categories, setCategories] = useState( defaultCategories );


  return (
    <>
        <h2>GifExpertApp</h2>
        <AddCattegory setCategories={ setCategories }/>
        <hr/>
        <ol>
            { 
                categories.map( category =>
                    <GifGrid 
                        category={ category }
                        key = { category }
            />)}
        </ol>
    </>
  )
}
