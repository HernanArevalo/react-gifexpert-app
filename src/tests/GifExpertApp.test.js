import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp"

describe('GifExpertApp tests', ()=>{

    test('snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe mostrar una lista de categorias', () => {

        const categories = ['The office', 'Brooklyn 99']

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )

    })

});