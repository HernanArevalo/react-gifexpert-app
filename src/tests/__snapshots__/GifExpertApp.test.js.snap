// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCattegory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The office"
      key="The office"
    />
    <GifGrid
      category="Brooklyn 99"
      key="Brooklyn 99"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp tests snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCattegory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
